src/index.ts

```diff
@@ -82,6 +95,69 @@ function tryParseFloat(value: string) {
 	return result;
 }
 
+function normalizeNumber(value: string, allowDot: boolean = true): string {
```

Comment:
I was able to crunch this function down to the following without breaking tests:
```ts
function normalizeNumber(value: string, allowDot = true) {
	const match = numberPrefixPattern.exec(value);
	if (!match) {
		throw new Error(`Invalid number: ${value}`);
	}
	const [, sign, dot] = match;
	if (sign === '+') {
		value = value.substr(1);
	}
	if (dot) {
		if (!allowDot) {
			throw new Error(`Invalid number (too many dots): ${value}`);
		}
		if (sign === '-') {
			value = '-0' + value.substr(1);
		} else {
			value = '0' + value;
		}
	}
	return (dot || countDots(value))
		? value.replace(/\.?0+$/, '')
		: value;
}
```

Reply:
yes -- the previous version did focus on computing as little as possible, e.g. run `countDots()` only if necessary, and also apply `replace(/\.?0+$/, '')` only on input where it would have an effect

but the gain of that is probably negligible

Reply:
Yeah and it still has to do the check before doing the replace, so it probably ends up being about the same performance there.